### PR TITLE
[#123479051] Sanger sequencing deleting a batch

### PR DESCRIPTION
**Pull request by jhanggi**
Extends from #674 and #673.

This also adds authorization around Batch actions and locks them to a facility.

### CODE REVIEW

**Comment by drench** [comment]
The only issue I've run into while testing is my admin tabs disappearing when under Sanger; all I get are Home, Sanger, and Manage Facilities.

**Comment by jhanggi** [comment]
Just pushed fixes for the navigation issue.

File: vendor/engines/sanger_sequencing/app/views/sanger_sequencing/admin/batches/index.html.haml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  %thead
+    %tr
+      %th
+      %th= SangerSequencing::Batch.model_name.human
+      %th= SangerSequencing::Batch.human_attribute_name(:created_at)
+      %th= SangerSequencing::Batch.human_attribute_name(:created_by)
+      %th= SangerSequencing::Batch.human_attribute_name(:submissions)
+      %th= SangerSequencing::Batch.human_attribute_name(:samples)
+  %tbody
+    - @batches.each do |batch|
```
Comment (drench) [inline vendor/engines/sanger_sequencing/app/views/sanger_sequencing/admin/batches/index.html.haml]: Any plans for an `if @batches.empty?` case?

File: vendor/engines/sanger_sequencing/app/views/sanger_sequencing/admin/shared/_sidenav.html.haml [hunk 2]
```diff
@@ -1,2 +1,3 @@
 %ul.sidebar-nav
   = tab SangerSequencing::Submission.model_name.human(count: 2), facility_sanger_sequencing_admin_submissions_path(current_facility), sidenav_tab == "submissions"
+  = tab SangerSequencing::Batch.model_name.human(count: 2), facility_sanger_sequencing_admin_batches_path(current_facility), sidenav_tab == "batches"
```
Comment (drench) [inline vendor/engines/sanger_sequencing/app/views/sanger_sequencing/admin/shared/_sidenav.html.haml]: With training requests we put a count in the sidenav. Would something like that be useful for submissions or batches?

**Comment by drench** [comment]
The interface works well for me. I had a couple questions/suggestions.

File: vendor/engines/sanger_sequencing/app/views/sanger_sequencing/admin/batches/index.html.haml (hunk 1 above)
Comment (jhanggi) [inline vendor/engines/sanger_sequencing/app/views/sanger_sequencing/admin/batches/index.html.haml]: That's probably a good idea. I'll add that in.

File: vendor/engines/sanger_sequencing/app/views/sanger_sequencing/admin/shared/_sidenav.html.haml (hunk 2 above)
Comment (jhanggi) [inline vendor/engines/sanger_sequencing/app/views/sanger_sequencing/admin/shared/_sidenav.html.haml]: Hmm, that's an interesting idea. I'm not quite sure whether that'll be useful or not. Maybe once we have a little more in the way of statuses on batches. Let's keep in mind as we roll out the feature.

**Comment by jhanggi** [comment]
Rebased off latest master and handled the "none" case for both submissions and batches.

**Comment by drench** [comment]
:+1:

**Pull request merged**